File: src/main/java/dev/mayaqq/estrogen/integrations/EarsCompat.java
package dev.mayaqq.estrogen.integrations;

import com.unascribed.ears.api.EarsFeatureType;
import com.unascribed.ears.api.registry.EarsInhibitorRegistry;
import dev.mayaqq.estrogen.registry.EstrogenEffects;
import net.minecraft.entity.player.PlayerEntity;

public class EarsCompat {
    public static void boob() {
        EarsInhibitorRegistry.register("estrogen", (part, peer) -> part == EarsFeatureType.CHEST && ((PlayerEntity)peer).hasStatusEffect(EstrogenEffects.ESTROGEN_EFFECT));
    }
}

File: src/common/main/kotlin/dev/mayaqq/estrogen/compat/ears/EarsCompat.kt
package dev.mayaqq.estrogen.compat.ears

import com.unascribed.ears.api.EarsFeatureType
import com.unascribed.ears.api.registry.EarsInhibitorRegistry
import dev.mayaqq.estrogen.client.features.boobs.Boob.shouldShow
import dev.mayaqq.estrogen.config.EstrogenClientConfig
import dev.mayaqq.estrogen.injection.chestConfig
import net.minecraft.world.entity.player.Player


object EarsCompat {
    fun boob() {
        EarsInhibitorRegistry.register(
            "estrogen"
        ) { part: EarsFeatureType, peer: Any ->
                EstrogenClientConfig.Compat.ears &&
                part == EarsFeatureType.CHEST &&
                shouldShow(peer as Player) &&
                EstrogenClientConfig.ChestRenderingGlobal.rendering &&
                peer.chestConfig != null && peer.chestConfig!!.enabled
        }
    }
}